arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -93,7 +93,12 @@ auto DocumentFollowerState::getAssociatedShardList() const
   std::vector<ShardID> shardIds;
   shardIds.reserve(collections.size());
   for (auto const& shard : collections) {
-    shardIds.emplace_back(shard->name());
+    auto maybeShardId = ShardID::shardIdFromString(shard->name());
+    ADB_PROD_ASSERT(maybeShardId.ok())
+        << "Tried to produce shard list on Database Server for a collection "
+           "that is not a shard "
+        << shard->name();
+    shardIds.emplace_back(std::move(maybeShardId.get()));
```

Comment:
do we need the extra std::move here?
```suggestion
    shardIds.emplace_back(std::move(maybeShardId.get()));
```

Reply:
No, we do not, removed